go/private/path.bzl

```diff
@@ -0,0 +1,101 @@
+# Copyright 2014 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_go//go/private:providers.bzl", "GoLibrary", "GoPath")
+
+def _go_path_impl(ctx):
+  print("""
+EXPERIMENTAL: the go_path rule is still very experimental
+Please do not rely on it for production use, but feel free to use it and file issues
+""")
+  # First gather all the library rules
+  golibs = depset()
+  for dep in ctx.attr.deps:
+    golib = dep[GoLibrary]
+    golibs += [golib]
+    golibs += golib.transitive
+  seen_libs = {}
```

Comment:
Nit: move these two lines below the comment, since they are part of the next logical step.